plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1756,359 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
+            assertQuery(session, "SELECT a.x, b.x from %s a JOIN %s b on (a.part = b.part) where a.part = 'part_a'".formatted(table1.getName(), table2.getName()), "VALUES ('a', 'a')");
+        }
+    }
+
+    @Test
+    public void testPartitionPredicateAllowed()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner(), session),
+                "test_partition_predicate",
+                "(x VARCHAR, part VARCHAR) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("('a', '1')", "('b', '2')"))) {
+            assertQuery(session, "SELECT * FROM " + table.getName() + " WHERE CAST (part AS INTEGER) = 1", "VALUES ('a', 1)");
+        }
+    }
+
+    @Test
+    public void testPartitionPredicateInOuterQuery()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner(), session),
+                "test_partition_predicate",
+                "(x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("(1, 11)", "(2, 22)"))) {
+            assertQuery(session, "SELECT * FROM (SELECT * FROM " + table.getName() + " WHERE x = 1) WHERE part = 11", "VALUES (1, 11)");
+        }
+    }
+
+    @Test
+    public void testPartitionPredicateInInnerQuery()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner(), session),
+                "test_partition_predicate",
+                "(x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("(1, 11)", "(2, 22)"))) {
+            assertQuery(session, "SELECT * FROM (SELECT * FROM " + table.getName() + " WHERE part = 11) WHERE x = 1", "VALUES (1, 11)");
+        }
+    }
+
+    @Test
+    public void testPartitionPredicateAnalyze()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner(), session),
+                "test_partition_predicate_analyze_",
+                "(x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("(1, 11)", "(2, 22)"))) {
+            @Language("RegExp")
+            String expectedMessageRegExp = "Filter required on test_schema\\." + table.getName() + " for at least one partition column: part";
+            assertQueryFails(session, "ANALYZE " + table.getName(), expectedMessageRegExp);
+            assertQueryFails(session, "EXPLAIN ANALYZE " + table.getName(), expectedMessageRegExp);
```

Comment:
`ANALYZE` statement on partitioned tables no longer works when `query_partition_filter_required` is enabled. Is it fine? Hive connector supports `partitions` analyze properties, but Delta Lake connector doesn't.